usr/lib/pkcs11/common/mech_sha.c

```diff
@@ -38,356 +38,23 @@
 #include "tok_spec_struct.h"
 #include "trace.h"
 
-#define SHA_HARDWARE_THRESHHOLD 128000
+#include <openssl/sha.h>
```

Comment:
Same comment regarding evp as in the aes case applies here. However, for sha1 old interfaces have hw support on z.

Reply:
Also not performance relevant. In general the token provided SHA routines are used. Only if the token does not provide a SHA routine, and its SHA1, then it falls back to the SW implementation. All our tokens do provide SHA routines.

compute_sha1 is used for pin verification, private token object handling and master key handling only. 

compute_sha is used inside mech_rsa for emsa_pss encode and verify, and oaep. Only if SHA1 is used and the token does not provide a SHA implementation, then the SW fallback is used.

Reply:
alright